base/src/main/java/uk/ac/ebi/atlas/experimentimport/idf/IdfParser.java

```diff
@@ -35,16 +39,11 @@ public IdfParser(IdfStreamerFactory idfReaderFactory) {
     public IdfParserOutput parse(String experimentAccession) {
         try (TsvStreamer idfStreamer = idfReaderFactory.create(experimentAccession)) {
 
-            parsedIdf =
-                    idfStreamer.get()
+            parsedIdf = idfStreamer.get()
                             .filter(line -> line.length > 1)
-                            .filter(line -> INVESTIGATION_TITLE_ID.equalsIgnoreCase(line[0].trim()) ||
-                                        AE_EXPERIMENT_DISPLAY_NAME_ID.equalsIgnoreCase(line[0].trim()) ||
-                                        PUBMED_ID.equalsIgnoreCase(line[0].trim()) ||
-                                        PUBLICATION_TITLE_ID.equalsIgnoreCase(line[0].trim()) ||
-                                        EXPECTED_CLUSTERS_ID.equalsIgnoreCase(line[0].trim()))
+                            .filter(line -> LINE_IDS.contains(line[0].trim().toUpperCase()))
                             .collect(Collectors.toMap(
-                                    line -> line[0].toUpperCase().trim(),
+                                    line -> line[0].trim().toUpperCase(),
```

Comment:
Maybe you can `.map(line -> line[0].toUpperCase().trim())` and then filter/collect?

Reply:
I'm not sure what you mean. I am building a map with `line[0]` as key and `line[1:]` as value. If I map the whole thing to `line[0].toUpperCase().trim()` then I'll lose the rest of the line contents...